src/main/java/step1/StringCalculator2.java

```diff
@@ -0,0 +1,109 @@
+package step1;
+
+import java.util.Stack;
+
+public class StringCalculator2 {
+
+    private final static String SEPARATOR = " ";
+    private final static String[] OPERATORS = {"+", "-", "*", "/"};
+    private String[] requests;
+    private Stack<Integer> numberStack = new Stack<>();
+    private Stack<String> operatorStack = new Stack<>();
+    private int result = 0;
+    private String currentOperator = null;
+
+    public int execute(String input) {
+        inputValidate(input);
+
+        requests = input.split(SEPARATOR);
+        for (int i = requests.length - 1; i >= 0; i--) {
+            separate(i);
+        }
+        return calculate();
+    }
+
+    public void inputValidate(String input) {
+        if (input == null || input.length() == 0) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    public void operatorValidate(String operator) {
+        for (int i = 0; i < OPERATORS.length; i++) {
+            if (!OPERATORS[i].contains(operator)) {
+                throw new IllegalArgumentException();
+            }
+        }
+    }
+
+    private int calculate() {
+        result = numberStack.pop();
+        for (; !operatorStack.empty(); ) {
+            currentOperator = operatorStack.pop();
+            operatorValidate(currentOperator);
+            addOperatorCheck();
+            subtractOperatorCheck();
+            multiplyOperatorCheck();
+            divideOperatorCheck();
+        }
+        return result;
+    }
+
+    private void addOperatorCheck() {
+        if (currentOperator.equals("+")) {
```

Comment:
만약 `currentOperator`가 `null`이라면 어떤 위험이 발생할까요? 그 위험을 피할 수 있는 방법도 찾아보면 좋을 것 같아요. 아래의 글이 도움 될 거예요.
https://www.javacodegeeks.com/2012/06/avoid-null-pointer-exception-in-java.html

Reply:
if (! currentOperator.isEmpty() && "+".equals(currentOperator))
이런 느낌으로 구현하는게 맞나요~?
리뷰어님이 힌트 주신 방법으로 로직을 개선하다보니 if 문이 없어졌어요!! 👍

Reply:
`currentOperator`가 `null`이라면 `currentOperator.isEmpty()`도 위험합니다. 왜 그럴까요?

```suggestion
        if ("+".equals(currentOperator)) {
```

Reply:
```currentOperator``` 가 ```null``` 일 시 NPE가 발생하기 때문에 런타임 에러가 떨어질 것 같습니다!!
isEmpty 뿐만아니라 null 체크도 해줘야 한다고 생각합니다. Optional을 활용하면 이 부분은 개선될 수 있을까요??